telosnetwork/telos-wallet
# #368 |  NFT Inventory Filtering

# Fixes #368 

## Description
This PR adds the ability to filter NFT inventory based on collection (minting contract) and search terms.

This PR also updates the 'back' button behavior to be a bit more sensible. When the user has come from another site, such as deeplinking to the Send page, the back button will navigate to the parent page. when the user has come from another page on the site, such as going to the NFT detail page from the inventory page, the back button actually goes back in history to preserve query params from the previous page (e.g. to maintain pagination in the NFT inventory table)

## Test scenarios
Note: this will need to be tested locally on mainnet
- from the project root, run `git fetch --all && git checkout 368-inventory-filtering`
- change `.env` to say `NETWORK="mainnet"`
- open `nfts.ts` and change line 126 to ` const owner = '0x13B745FC35b0BAC9bab9fD20B7C9f46668232607';` which is an account that has NFTs on mainnet
- run `yarn dev`
- go to http://localhost:8081/evm/collectible-inventory
- click the `Collection` dropdown
    - you should see a list of collections sorted alphabetically at the top (for contracts with known names), then at the bottom, shortened contract address(es)
- click a Collection
    - the inventory should filter based on that collection
- clear the collection
    - inventory should go back to showing all, at the first page
- enter a search term in the search bar
    - the inventory should filter based on that query (see Issue for details on what is searched upon)
- clear the search term
    - inventory should go back to showing all, at the first page 
- try different combinations of collection and search terms
    - results should show as expected
    - there should be a No Results state when appropriate
- try different combinations of pagination (change page, change rows per page) and collection and search
    - results should work correctly
- as you try combinations, try clicking on an NFT to go to the detail page, then navigate back
    - your position in the table should persist



## Checklist:
